### PR TITLE
Update to `cargo-dist` v0.23.0

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

Closes https://github.com/astral-sh/uv/issues/8242.

### CODE REVIEW

**Comment by charliermarsh** [comment]
Need to add some docs, this adds a few new env vars.

**Review by zanieb — APPROVED** [review]

File: docs/getting-started/installation.md [hunk 1]
```diff
@@ -83,6 +83,13 @@ options can be passed directly to the install script. For example, to see the av
 $ curl -LsSf https://astral.sh/uv/install.sh | sh -s -- --help
 ```
 
+In ephemeral environments like CI, use `UV_UNMANAGED_INSTALL` to install uv to a specific path while
+preventing the installer from modifying shell profiles or environment variables:
```
Comment (zanieb) [inline docs/getting-started/installation.md:87]: Also?

> Disables updater-related tooling, including install receipt creation
> Forces a flat installation layout, installing all files into a single directory

Comment (charliermarsh) [reply docs/getting-started/installation.md:87]: Yeah

Comment (charliermarsh) [reply docs/getting-started/installation.md:87]: (Added a note.)

**Pull request merged**